src/Resources/contao/library/Contao/Widget.php

```diff
@@ -746,11 +752,28 @@ public function getAttribute($strKey)
 
 
 	/**
+	 * By default, Widget will use its getPost() method when validate() is
+	 * called. This method allows you to fetch the value from elsewhere.
+	 *
+	 * @param callable $callable
+	 *
+	 * @return $this
+	 */
+	public function setInputCallback(callable $callback = null)
+	{
+		$this->inputCallback = $callback;
+
+		return $this;
+	}
+
+
+	/**
 	 * Validate the user input and set the value
 	 */
 	public function validate()
 	{
-		$varValue = $this->validator($this->getPost($this->strName));
+        $varValue = (null !== $this->inputCallback ? call_user_func($this->inputCallback) : $this->getPost($this->strName));
```

Comment:
Shouldn't this rather be:
```diff
-		$varValue = $this->validator($this->getPost($this->strName));
+        $varValue = (is_callable($this->inputCallback) ? call_user_func($this->inputCallback) : $this->getPost($this->strName));
```
IMO this will prevent errors when wrong values get passed or a class is not loadable or the like.
What do you think?